modules/CbChargeSOMDriver/evse_board_support/evse_board_supportImpl.cpp

```diff
@@ -0,0 +1,447 @@
+        return {types::board_support_common::Event::D};
+    case types::cb_board_support::CPState::E:
+        return {types::board_support_common::Event::E};
+    case types::cb_board_support::CPState::F:
+        return {types::board_support_common::Event::F};
+    default:
+        throw std::runtime_error("Unable to map the value '" + cpstate_to_string(other) + "'.");
+    }
+}
+
+namespace module {
+namespace evse_board_support {
+
+void evse_board_supportImpl::init() {
+    // configure hardware capabilities
+    this->hw_capabilities.max_current_A_import = 63;
+    this->hw_capabilities.min_current_A_import = 0;
+    this->hw_capabilities.max_phase_count_import = 3;
+    this->hw_capabilities.min_phase_count_import = 1;
+    this->hw_capabilities.max_current_A_export = 63;
+    this->hw_capabilities.min_current_A_export = 0;
+    this->hw_capabilities.max_phase_count_export = 3;
+    this->hw_capabilities.min_phase_count_export = 1;
+    this->hw_capabilities.supports_changing_phases_during_charging = false;
+    this->hw_capabilities.connector_type =
+        types::evse_board_support::string_to_connector_type(this->mod->config.connector_type);
+
+    // register our callback handlers
+    this->mod->controller.signal_pp_state_change.connect(&evse_board_supportImpl::pp_observation_worker, this);
+    this->mod->controller.signal_cp_state_change.connect(&evse_board_supportImpl::cp_observation_worker, this);
+}
+
+void evse_board_supportImpl::ready() {
+    // the BSP must publish this variable at least once during start up
+    this->publish_capabilities(this->hw_capabilities);
+}
+
+void evse_board_supportImpl::update_cp_state_internally(types::cb_board_support::CPState state,
+                                                        const struct cp_state_signal_side& negative_side,
+                                                        const struct cp_state_signal_side& positive_side) {
+    if (this->cp_current_state == state)
+        return;
+
+    EVLOG_info << "CP state change from " << this->cp_current_state << " to " << state << ", "
+               << "U_CP+: " << positive_side.voltage << " mV, "
+               << "U_CP-: " << negative_side.voltage << " mV";
+    this->cp_current_state = state;
+}
+
+void evse_board_supportImpl::handle_enable(bool& value) {
+    // generate state A or state F
+    double new_duty_cycle = value ? 100.0 : 0.0;
+    // pause CP observation to avoid race condition between this thread and the CP observation thread
+    this->disable_cp_observation();
+    EVLOG_info << "handle_enable: Setting new duty cycle of " << std::fixed << std::setprecision(2) << new_duty_cycle
+               << "%";
+    do {
+        try {
+            this->mod->controller.cp_set_duty_cycle(new_duty_cycle);
+            break;
+        } catch (std::system_error& e) {
+            EVLOG_error << e.what();
+
+            // retry after a short period of waiting
+            std::this_thread::sleep_for(100ms);
```

Comment:
I know it's a temporary solution, but have a single define/constant for the repeat time would be great.